src/FormlyMapper/FormlyField/BirthdayField.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Linio\DynamicFormBundle\FormlyMapper\FormlyField;
+
+use Linio\DynamicFormBundle\FormlyMapper\FormlyField;
+
+class BirthdayField extends FormlyField
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function getFieldType()
+    {
+        return 'birthday';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function buildFieldTypeConfiguration(): void
+    {
+        $this->formlyFieldConfiguration['templateOptions']['type'] = $this->getFieldType();
+
+        $order = 'desc';
+        if(isset($this->fieldConfiguration['options']['order'])) {
+            $tmpOrder = strtolower($this->fieldConfiguration['options']['order']);
+            if($tmpOrder == 'asc' || $tmpOrder == 'desc') {
+                $order = $tmpOrder;
+            }
+            unset($this->formlyFieldConfiguration['templateOptions']['order']);
+        }
+
+        if(isset($this->fieldConfiguration['options']['minAgeAllowed']) && isset($this->fieldConfiguration['options']['maxAgeAllowed'])) {
+            $minAgeAllowed = $this->fieldConfiguration['options']['minAgeAllowed'];
+            $maxAgeAllowed = $this->fieldConfiguration['options']['maxAgeAllowed'];
+            if(is_numeric($minAgeAllowed) && ($minAgeAllowed >= 0) && is_numeric($maxAgeAllowed) && ($maxAgeAllowed >= 0)) {
+                if($minAgeAllowed > $maxAgeAllowed) {
+                    $tmp = $minAgeAllowed;
+                    $minAgeAllowed = $maxAgeAllowed;
+                    $maxAgeAllowed = $tmp;
+                }
+                
+                $yearsRange = range(date('Y')-$minAgeAllowed, date('Y')-$maxAgeAllowed);
+
+                $this->formlyFieldConfiguration['templateOptions']['years'] = ($order == 'asc') ? array_reverse($yearsRange) : $yearsRange;
+            }
```

Comment:
```php

$options = $this->fieldConfiguration['options']; 

// Is better for redability to use early returns, "return early, return often" https://blog.humphd.org/vocamus-1421/
if(!isset($options['minAgeAllowed']), $options['maxAgeAllowed'])
     || !is_numeric($options['minAgeAllowed']) || !is_numeric($options['maxAgeAllowed']
     || !$options['minAgeAllowed'] >=0  ||  !$options['maxAgeAllowed'] >=0 )
{
    return;
 }
 
$minAgeAllowed =  min($options['minAgeAllowed'], $options'['maxAgeAllowed']);
$maxAgeAllowed = max($options['minAgeAllowed'], $options'['maxAgeAllowed']);

$yearsRange = range(date('Y')-$minAgeAllowed, date('Y')-$maxAgeAllowed);

$this->formlyFieldConfiguration['templateOptions']['years'] = ($order == 'asc') ? array_reverse($yearsRange) : $yearsRange;

```

This is pseudo-code, I haven't tested it, but I think you can get the idea from this.